app/adapters/signal_adapter/outbound.rb

```diff
@@ -38,7 +38,7 @@ def perform(text:, recipient:)
       end
       res.value # may raise exception
     rescue Net::HTTPServerException => e
-      ErrorNotifier.report(SignalError.new, context: {
+      ErrorNotifier.report(ServerException.new, context: {
                              code: e.response.code,
```

Comment:
Why do you create a new exception object here instead of simply reporting the rescued exception? We lose the stack trace of the original exception… :(

```suggestion
      ErrorNotifier.report(e, context: {
```

Reply:
Ok, I tested this out and I think it's not actually true that it is easier to see where the error occurred because with the original exception you also have the class name on the high level view. See screenshot

![Screenshot from 2022-06-28 09-38-53](https://user-images.githubusercontent.com/26943915/176121461-225e806e-b2ee-4a80-970a-9e282b7db973.png)

That being said, I don't think we actually lose anything as you suggested. When I reported the original exception, it was just a string `400 "Bad Request"`, which we have in the code/message.

Compare the two screenshots:


![Screenshot from 2022-06-28 09-38-17](https://user-images.githubusercontent.com/26943915/176121786-e7aa77d4-ed68-40e2-b965-0b8af94de5fb.png)
![Screenshot from 2022-06-28 09-21-36](https://user-images.githubusercontent.com/26943915/176121861-cafa8e7e-ea8a-4f15-bd7a-32c79df82f60.png)

Reply:
Ah, sorry, I was actually talking about the stack trace which is empty, but I can see it’s there for the recent reported events! :)

![image](https://user-images.githubusercontent.com/1512805/178522234-ec159b6b-e69a-4280-881c-5a773ecf9b02.png)